### DISCUSSION TITLE
Improving your Sentry Debugging Experience with Insights into Feature Flags

### DISCUSSION THREAD

**Discussion by jas-kas** [getsentry/sentry#73542] [Problem Understanding]
### 📄 Background

We, at Sentry, use feature flags regularly to change our application behaviour and appearance conveniently without deploying new code. For example, our development process includes slowly rolling out new features to ensure they don’t introduce new errors and collect early product feedback.

We know that feature flag changes are just as valuable and critical to the debugging process as are code releases for some organizations. As such, we're looking at enhancing our context to include insights on your feature flags and integrating with popular feature flag systems. For example, displaying flags evaluated at the time of an error event, in order, on our Issue Details page.

###  👋 We’d like to hear from you

- How important are feature flags in your development process and how do you use them?
- What feature flagging system are you using today?
- When using Sentry to debug issues, what information would you want from your feature flag system there and how would you want it presented?
- What do you monitor to know if it is safe to continue to rollout a feature flag, and how can Sentry help facilitate this process?
- What kind of issues have you encountered that were related to feature flag changes and how did you go about debugging them today? How often do those types of issues happen?

Please share your initial thoughts and impressions below. 👇 If you’re interested in chatting live with the Sentry team on this topic, let us know below and we can schedule a call. Thank you!

**Comment by volokluev** [comment]
> When using Sentry to debug issues, what information would you want from Flagpole [or insert other feature flag system] there?

Snuba has its own runtime config that reads from redis. Knowing the feature flags that were toggled at what time could be good for sure

> How do you monitor if it's safe to continue to rollout a feature flag?

At this time I am switching between datadog for understanding infra load and sentry for business logic issues

> What kind of issues have you encountered that were related to feature flag changes and how did you go about debugging them today?

Often, there's something in the sentry repo that is changed (by turning on a feature flag) that will cause a lot of downstream impacts on the infra. At the moment the way I debug them is go into [#team-engineering](https://sentry.slack.com/archives/C1B4LB39D) and ask "yo what's going on, who broke this"

**Comment by sandstrom** [comment]
Some feedback on this new beta feature:

- We have feature flags internally, so would be nice if we could just tell the JS and Ruby SDK which ones are active during a particular error.
- We can enable them both globally, on the account-level (we're a B2B SaaS app) or on the user-level. All of these doesn't have to be reflected in Sentry, but just wanted to explain how we use them.
- We also have a list of globally available flags (distinct from enabled), i.e. the total set of flags available at any time. But not sure this is very useful information.

**Reply by aliu39** [reply]
> would be nice if we could just tell the JS and Ruby SDK which ones are active during a particular error.

We recently added a JS integration letting you manually track your FF state!  See https://docs.sentry.io/platforms/javascript/configuration/integrations/featureflags/

**Reply by sandstrom** [reply]
@aliu39 Great! 🎉 

Is there an issue tracking it's addition to the Ruby SDK?

**Reply by aliu39** [reply]
No, but feel free to make one. https://develop.sentry.dev/sdk/expected-features/#feature-flags describes the expected features for this kind of integration.

**Reply by moredip** [reply]
Hi 👋 

I just discovered that Sentry integrates with OpenFeature - great to see!

Any objections to us adding the integration to our [Ecosystem page](https://openfeature.dev/ecosystem)?

**Reply by jas-kas** [reply]
@moredip No objections at all - please add 😄

**Reply by michellewzhang** [reply]
Hi @moredip, Sentry also has a Python hook that isn't documented yet -- I added it here: https://github.com/open-feature/openfeature.dev/pull/1050

I wasn't sure how to request a reviewer, but let me know if there are any problems with this!

**Comment by stwr667** [comment]
Well done for starting this last year 👏 🙌 . Our organisation is wanting to use this feature, because we now _**more often**_ release features with feature flag rollouts, rather than code deployments. Because of this, Sentry is fast becoming ineffective at being able to clearly link new errors to the code that introduced it.

Questions
1. Given that, are there any more features roadmapped in the Feature Flag space, and any to be shipped soon?
2. And more specifically, [as per the docs here](https://docs.sentry.io/organization/integrations/feature-flag/statsig/#:~:text=We%20currently%20only%20support%20change%20tracking%20events%20from%20this%20webhook), Sentry doesn't currently support Evaluation Tracking for Statsig. Or for the Java language. Are there any plans to support either of them?
3. And lastly, the feature is currently marked as [Beta](https://docs.sentry.io/product/issues/issue-details/feature-flags/). When will it be fully supported?

Suggestion
1. Our organisation (~15,000 staff) has 1 main Statsig site, and 1 Sentry site. But there are many teams & projects in each. For now, it seems like the integration would mean that e.g. a single Sentry project would see the feature flag changes of ~15,000 staff on a single Sentry issue, when really, it's part of a Sentry project that only 50-100 people use. So is there a plan to allow user-defined mappings of FeatureFlag projects to Sentry projects? Or some way to reduce the scope & noise?

Thank you!

**Reply by aliu39** [reply]
Hi @stwr667 , Statsig eval tracking is supported for [javascript and python](https://docs.sentry.io/organization/integrations/feature-flag/statsig/#set-up-evaluation-tracking). Hope I understood question 2 right. As for java support I'm not sure what the timeline is

**Reply by jas-kas** [reply]
Hey @stwr667, 

Thanks for reaching out!

To answer 1), we're looking to expand our feature flag support to all core mobile platforms (React Native, Android, iOS) as feature flagging is very common in mobile development. That is what we have roadmapped.

For 3) we don't have a timeline for this but I expect it to be moved out of beta once we have parity across all core web and mobile platforms.
